rllib/evaluation/rollout_worker.py

```diff
@@ -1561,7 +1561,7 @@ def set_global_vars(self, global_vars: dict) -> None:
         Examples:
             >>> global_vars = worker.set_global_vars({"timestep": 4242})
         """
-        self.foreach_policy(lambda p, _: p.on_global_var_update(global_vars))
+        self.foreach_policy_to_train(lambda p, _: p.on_global_var_update(global_vars))
```

Comment:
I see. so this PR doesn't really change what policies get loaded to the map, it only changes what policies get updated. is my understanding correct?
thanks.

Reply:
Yes, this change only changes what policy is updated with global vars.

The problem: 
- In order to update a policy, if it isn't in the memory cache it will need to be loaded from disk.
- Looking through the project, it looks like every time `set_global_vars()` is called `_get_global_vars` is passed as argument
- I'm not sure what kind of global vars there usually are, but looking into `rllib/execution/common.py` it looks like `_get_global_vars()` basically always returns an updated timestep
- Consequently, if any policy is written to disk, these are read from disk all the time to update their timesteps and we have constant disk access again, slowing everything down, kind of defeating the reason why we put them to the disk in the first place.

The solution:
- What at least helps with this problem is only updating global vars for policies that are being trained.
- If all policies that are being trained fit into the in-memory cache, the disk won't constantly be accessed.
- If they don't fit, then I see no way to stop constant disk access anyway.
- I'm not 100% sure this doesn't have other consequences, someone with better knowledge of the code base would have to judge that. However, I see no reason why policies that are not being trained would need their vars updated (are there even other vars besides timesteps).
- At any rate, without this case we are again at a point where any disk that is being offloaded to disk is constantly being accessed, making it basically unusable to have more policies than the `policy_map_capacity`.